app/controllers/products_controller.rb

```diff
@@ -32,10 +33,20 @@ def index
   def available_for_cross_core_ordering
     respond_to do |format|
       format.js do
-        @facility_products = current_facility.products.mergeable_into_order.cross_core_available.alphabetized.map { |p| { id: p.id, name: p.name, time_based: p.order_quantity_as_time? } }
-
         render json: @facility_products
       end
     end
   end
+
+  def set_facility_products
+    original_order_facility = params[:original_order_facility]&.to_i
+
+    query = current_facility.products.mergeable_into_order
+
+    unless original_order_facility.present? && (current_facility.id == original_order_facility)
+      query = query.cross_core_available
+    end
```

Comment:
Is this variable name accurate/helpful?  I think a variable name or code comment would help future developers reason about this:
```suggestion
    adding_to_order_in_original_facility = original_order_facility.present? && (current_facility.id == original_order_facility)
    query = query.cross_core_available if adding_to_order_in_original_facility
```